cpp/src/aztec/honk/utils/power_polynomial.hpp

```diff
@@ -0,0 +1,65 @@
+#include <polynomials/polynomial.hpp>
+#include <common/max_threads.hpp>
+
+#ifndef NO_MULTITHREADING
+#include "omp.h"
+#endif
+namespace honk {
+namespace power_polynomial {
+/**
+ * @brief Generate the power polynomial vector
+ *
+ * @details Generates a vector, where v[i]=ζⁱ
+ *
+ * @param zeta
+ * @param vector_size
+ * @return barretenberg::polynomial
+ */
+barretenberg::polynomial generate_vector(barretenberg::fr zeta, size_t vector_size)
+{
+    barretenberg::polynomial pow_vector(vector_size);
```

Comment:
If the size does not grow, consider `pow_vector(vector_size, vector_size);` so that we don't over-allocate.